engine/src/main/java/org/terasology/rendering/dag/nodes/OpaqueBlocksNode.java

```diff
@@ -122,15 +125,39 @@ public OpaqueBlocksNode(Context context) {
     }
 
     public void enableWireframe() {
+        boolean refreshTaskList = false;
+
+        EnableFaceCulling faceCullingStateChange = new EnableFaceCulling();
+        if (getDesiredStateChanges().contains(faceCullingStateChange)) {
+            removeDesiredStateChange(faceCullingStateChange);
+            refreshTaskList = true;
+        }
+
         if (!getDesiredStateChanges().contains(wireframeStateChange)) {
             addDesiredStateChange(wireframeStateChange);
+            refreshTaskList = true;
+        }
+
+        if (refreshTaskList) {
             worldRenderer.requestTaskListRefresh();
         }
     }
 
     public void disableWireframe() {
+        boolean refreshTaskList = false;
+
+        EnableFaceCulling faceCullingStateChange = new EnableFaceCulling();
+        if (!getDesiredStateChanges().contains(faceCullingStateChange)) {
```

Comment:
Similar comment to the one above: when you disable wireframe you can rely on the EnableFaceCulling state change to NOT be in the desired set because it has been removed by the enableWireframe method.

The only potential reason to do this here is if the wireframe mode is persisted between sessions. But in that case I'd put the IF in the constructor, not here. Can you verify?

Also similarly to the comment above, the IF block will always evaluate to true because the content of `faceCullingStateChange` is regenerated every time this method is invoked.

Reply:
Also notice OpaqueObjectsNode.java also needs the same treatment.